src/chage.c

```diff
@@ -231,10 +231,17 @@ static int new_fields (void)
 	return 1;
 }
 
-static void print_date (time_t date)
+static void print_day_as_date (long day)
 {
 	struct tm *tp;
 	char buf[80];
+	time_t date;
+
+	if (day < 0 || LONG_MAX / DAY < day) {
+		(void) puts (_("never"));
+		return;
+	}
+	date = (time_t) day * DAY;
```

Comment:
I suggest the following:

```c
if (day < 0) {
    puts(_("never"));
    return;
}
if (__builtin_mul_overflow(day, DAY, &date)) {
    puts(_("future"));
    return;
}
```

It's a bit simpler and more readable about checking for overflow, and it's also more consistent in the handling of overflow in dates compared to other functions:

```c
$ grepc date_to_str .
./lib/prototypes.h:extern void date_to_str (size_t size, char buf[size], long date);
./lib/date_to_str.c:void
date_to_str(size_t size, char buf[size], long date)
{
	time_t           t;
	const struct tm  *tm;

	t = date;
	if (date < 0) {
		(void) strtcpy(buf, "never", size);
		return;
	}

	tm = gmtime(&t);
	if (tm == NULL) {
		(void) strtcpy(buf, "future", size);
		return;
	}

	if (strftime(buf, size, "%Y-%m-%d", tm) == 0)
		(void) strtcpy(buf, "future", size);
}
```

Reply:
Oh, didn't see that "future" is actually used in other places. Adjusted.